test/AutoRest.TestServerLowLevel.Tests/lro.cs

```diff
@@ -220,44 +232,66 @@ public class LroTest : TestServerLowLevelTestBase
         {
             var operation = new LROsClient(Key, endpoint, null).DeleteAsyncRetrySucceeded(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
             return WaitForCompletion(operation, waitUntil);
         });
 
         [Test]
         public Task LRODeleteInlineComplete([Values(WaitUntil.Started, WaitUntil.Completed)] WaitUntil waitUntil) => TestStatus(async endpoint =>
         {
             var operation = await new LROsClient(Key, endpoint, null).Delete204SucceededAsync(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
             return await WaitForCompletionAsync(operation, waitUntil).ConfigureAwait(false);
         });
 
         [Test]
         public Task LRODeleteInlineComplete_Sync([Values(WaitUntil.Started, WaitUntil.Completed)] WaitUntil waitUntil) => TestStatus(endpoint =>
         {
             var operation = new LROsClient(Key, endpoint, null).Delete204Succeeded(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
             return WaitForCompletion(operation, waitUntil);
         });
 
         [Test]
         public Task LRODeleteNoHeaderInRetry([Values(WaitUntil.Started, WaitUntil.Completed)] WaitUntil waitUntil) => TestStatus(async endpoint =>
         {
             var operation = await new LROsClient(Key, endpoint, null).DeleteNoHeaderInRetryAsync(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
             return await WaitForCompletionAsync(operation, waitUntil).ConfigureAwait(false);
         });
 
         [Test]
         public Task LRODeleteNoHeaderInRetry_Sync([Values(WaitUntil.Started, WaitUntil.Completed)] WaitUntil waitUntil) => TestStatus(endpoint =>
         {
             var operation = new LROsClient(Key, endpoint, null).DeleteNoHeaderInRetry(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
             return WaitForCompletion(operation, waitUntil);
         });
 
+        [Test]
+        public Task LRODeleteValueInlineComplete([Values(WaitUntil.Started, WaitUntil.Completed)] WaitUntil waitUntil) => TestStatus(async endpoint =>
+        {
+            var operation = await new LROsClient(Key, endpoint, null).DeleteValue204SucceededAsync(waitUntil);
+            Assert.Throws<InvalidCastException>(() => { Operation<BinaryData> castedOperation = (Operation<BinaryData>)operation; });
+            return await WaitForCompletionAsync(operation, waitUntil).ConfigureAwait(false);
```

Comment:
Would it be possible to call `WaitForCompletion` on the operation directly, so we can see that the test is validating the correct APIs at a glance?

Reply:
Hi @annelo-msft  There is another synchronize test LRODeleteValueInlineComplete_Sync which will call WaitForCompletion. This Async test will call all async method.